js/src/figure.js

```diff
@@ -1384,44 +1252,82 @@ var FigureView = widgets.DOMWidgetView.extend( {
         d.object_ticklabel.parent.remove(d.object_ticklabel)
     },
     update_scatters: function() {
-        if(this.scatter_views) {
-            this.scatter_views.forEach((scatter) => {
-                scatter.remove_from_scene();
-                scatter.remove();
-            })
-        }
-        var scatters = this.model.get('scatters');
-        if(scatters) {
-            //this.scatters.update(scatters);
-            this.scatter_views = _.map(scatters, function(model) {
-                var options = {parent: this}
-                var scatter_view = new scatter.ScatterView({options: options, model: model})
-                scatter_view.render()
-                return scatter_view
+        var scatters = this.model.get('scatters'); // This is always a list?
+        if(scatters.length != 0) { // So now check if list has length 0
+            var current_sct_cids = []
+            // Add new scatter if not already as scatter view in figure
+            _.each(scatters, function(scatter_model) {
+                current_sct_cids.push(scatter_model.cid);
+                if(!(scatter_model.cid in this.scatter_views)){
+                    var options = {parent: this}
+                    var scatter_view = new scatter.ScatterView({options: options, model: scatter_model})
+                    scatter_view.render()    
+                    this.scatter_views[scatter_model.cid] = scatter_view
+                }
             }, this)
```

Comment:
Yea, because the previous one was just throw every view away when a new one is added. I just made this code to handle it a little bit better, but any improvements on this are welcome.

Reply:
I think this would work fine for now right? I mean the order is first iterate over the current meshes in the figure view and add them to an empty list of current cids and during iteration check if cid is already present as a view if not add it to the views. then after check with the current cids list if there is any view that is not in that list and then remove it. Or did you have any other ideas on this?